python/src/spark_rapids_ml/knn.py

```diff
@@ -916,6 +924,20 @@ class ApproximateNearestNeighbors(
         if set, algoParam is used to configure the algorithm, on each data partition (or maxRecordsPerBatch if Arrow is enabled) of the item_df.
         Note this class constructs the kmeans index independently on individual data partition (or maxRecordPerBatch if Arrow is enabled).
 
+        When algorithm is 'cagra':
+            parameters for index construction
+            * build_algo: (str, default = 'ivf_pq') algorithm to build graph index, can be either 'ivf_pq' or 'nn_descent'. nn_descent is expected to be generally faster than ivf_pq.
+            * intermediate_graph_degree: (int, default = 128) an intermediate variable used during graph index construction
+            * graph_degree: (int, default = 64) the degree of each node in the final graph index
+
+            parameters for search, full list in cuvs python API documentation.
+            * itopk_size: (int, default = 64) number of intermediate search results retained during the search. Larger value improves the search accuracy but increases the search time.
```

Comment:
I guess the intermediate topk related params need to be no smaller than the top level one?   Our code should enforce/set this.

Reply:
Yes, added validation on the itopk_size parameter. cuml increases the value to the nearest multiple of 32 and requires the value to be larger than or equal to k.